### PR TITLE
refactor(InlineLoading): add TypeScript types

### LINKED ISSUE #13562: Add TypeScript types to Loading, InlineLoading

### PR DESCRIPTION
**Pull request by alissa9090**
Closes #13562

Adds TypeScript types to InlineLoading component.
The 13562 issue is about Loading and InlineLoading components.
But it seems like Loading has been already converted into tsx. So this PR should close 13562.

Changelog

Added TypeScript types to InlineLoading component

### CODE REVIEW

**Review by guidari — APPROVED** [review]
LGTM 🚀

**Review by andreancardona — APPROVED** [review]
thank you for the contribution! 💯

File: packages/react/src/components/InlineLoading/InlineLoading.tsx [hunk 1]
```diff
@@ -12,15 +12,64 @@ import { CheckmarkFilled, ErrorFilled } from '@carbon/icons-react';
...
+const InlineLoading = ({
   className,
   status = 'active',
   iconDescription,
   description,
   onSuccess,
   successDelay = 1500,
-  ...other
-}) {
+  ...divProps
```
Comment (tay1orjones) [inline packages/react/src/components/InlineLoading/InlineLoading.tsx]: Is this rename necessary? It'd be ideal for this to follow the naming convention we have everywhere else

**Review by tay1orjones — APPROVED** [review]
